src/libs/actions/Task.js

```diff
@@ -439,6 +442,124 @@ function editTaskAndNavigate(report, ownerAccountID, {title, description, assign
     Navigation.dismissModal(report.reportID);
 }
 
+/**
+ * Update a task's assignee
+ *
+ * @param {Object} report
+ * @param {Number} ownerAccountID
+ * @param {Object} editedTask
+ * @param {String} editedTask.assignee
+ * @param {Number} editedTask.assigneeAccountID
+ * @param {Boolean} editedTask.isOptimisticAssignee
+ */
+function editTaskAssigneeAndNavigate(report, ownerAccountID, {assignee = '', assigneeAccountID = 0, isOptimisticAssignee = false}) {
+    // Create the EditedReportAction on the task
+    const editTaskReportAction = ReportUtils.buildOptimisticEditedTaskReportAction(currentUserEmail);
+
+    // If we make a change to the assignee, we want to add a comment to the assignee's chat
+    let optimisticAssigneeAddComment;
+    let assigneeChatReportID;
+    if (assigneeAccountID && assigneeAccountID !== report.managerID && assigneeAccountID !== ownerAccountID) {
+        assigneeChatReportID = ReportUtils.getChatByParticipants([assigneeAccountID])?.reportID;
+
+        if (assigneeChatReportID !== report.parentReportID.toString()) {
+            optimisticAssigneeAddComment = ReportUtils.buildOptimisticTaskCommentReportAction(
+                report.reportID,
+                report.reportName,
+                assignee,
+                assigneeAccountID,
+                `Assigned a task to you: ${report.reportName}`,
+            );
+        }
+    }
+
+    const optimisticData = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report.reportID}`,
+            value: {[editTaskReportAction.reportActionID]: editTaskReportAction},
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`,
+            value: {
+                managerID: assigneeAccountID || report.managerID,
+                managerEmail: assignee || report.managerEmail,
+            },
+        },
+    ];
+    const successData = [];
+    const failureData = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report.reportID}`,
+            value: {[editTaskReportAction.reportActionID]: {pendingAction: null}},
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`,
+            value: {assignee: report.managerEmail, assigneeAccountID: report.managerID},
```

Comment:
This is interesting... I see we also store `assignee` and `assigneeAccountID` in the report in `editTaskAndNavigate` but aren't those supposed to be stored in the `task` onyx key @thienlnam ?

Reply:
Looks like we do store them optimistically right before we call this function, so maybe we need to un-set them from the task on fail?

Reply:
Also here I'll probably change this to update `managerEmail` and `managerID` to old values